selector-set.js

```diff
@@ -342,7 +356,15 @@
       }
     }
 
-    this.size -= Object.keys(removedIds).length;
+    for (i in removedIds) {
+      delete selectorObjects[i];
+      this.size--;
+    }
+
+    this.selectors = selectors;
```

Comment:
I'd just directly do it here - a bit less confusing.
```suggestion
    this.selectors = [];
```